### ISSUE TITLE
[Mod] Privilege Areas

### ISSUE THREAD

**Issue by BiLachner** [zeuner/edutest#7]
Habe eben den Mod wiedergefunden. Soll innerhalb eines markierten Bereiches Rechte nehmen oder geben können. Kann mir viele Anwendungen dazu im Edu-Bereich vorstellen.
https://forum.minetest.net/viewtopic.php?f=11&t=5545&p=297809

Leider ist der Mod etwas alt und mit der neuesten Version gibt Fehlermeldungen.
Für den Fall, das mein Mann mal Zeit hat, könnte er es sich auch mal anschauen, aber er ist noch Modding-Anfänger :-)

**Comment by zeuner** [comment]
Sieht auf jeden Fall interessant und vielseitig aus. Ich habe den Code
kurz ueberflogen, und denke, dass man die Funktionalitaet auch mit der
aktuellen Minetest-Version zum Laufen bringt. Je nach
Anwendungsszenario koennte aber auch relevant sein, dass die
Rechtezuweisung nicht ganz "wasserdicht" zu sein scheint, da sie mit
bis zu 1 Sekunde Verzoegerung nach Ueberschreiten der Bereichsgrenzen
stattfindet.

Gibt es denn konkrete Anwendungsszenarien, die fuer eine Einbindung in
EDUtest relevant sind, also nicht ausschliesslich die
Weiterentwicklung/Aktualisierung der privilegeareas-Mod betreffen?

Originalnachricht:

**Comment by BiLachner** [comment]
Sorry, bin gerade mit vielen anderen Sachen beschäftigt.

Was die Verzögerung betrifft: entweder müsste man das wissen und entsprechend nichts Kritisches am Rand bauen oder "Privilege Areas" erweitert den Bereich automatisch um einen Block in alle Richtungen. Wobei ich ersteres bevorzugen würde.

Die Frage im letzten Absatz verstehe ich nicht? Ob man einfach "Privilege Areas" in EDUTest einbaut oder separat weiterentwickelt? ... ist das damit gemeint?

**Comment by zeuner** [comment]
Ich hatte nun etwas Zeit, mir das naeher anzusehen.

Mit diesem Patch konnte ich ohne Fehlermeldung eine Zone erzeugen und
bekam dann auch erwartungsgemaess das festgelegte Privileg zugewiesen:
[1].

Spontan faellt auf, dass man die Zonen ohne jegliche Privilegien
erzeugen kann. So kann sich ein Spieler selbst jedes Privileg
zuweisen, indem er vorher eine entsprechende Zone erzeugt.

Interessant waere bzgl. EDUtest, die Zonen mit Hilfe der visuellen
Markierungen zu erzeugen, die es fuer areas (Gebietsschutz) gibt.

[1] https://github.com/minetest-mods/privilegeareas/pull/3

Originalnachricht:

**Comment by zeuner** [comment]
Ja, ich denke auch, dass man meistens Loesungen finden wird, um mit
der Verzoegerung umzugehen.

Nein, es ging mir nicht darum, den privilegeareas-Code in EDUtest zu
uebernehmen. Ich halte eigentlich mehr davon, wenn Mods nach
Moeglichkeit eigenstaendig bleiben, damit von Weiterentwicklungen
nicht nur EDUtest-Nutzer profitieren.

Mir ging es vielmehr darum, wo es nuetzlich ist, zwischen beiden Mods
"Bruecken zu schlagen". privilegeareas ist bislang voellig unabhaengig
von EDUtest zu nutzen, bedient sich z.B. nicht anders, egal ob man auf
dem selben Server auch EDUtest installiert hat. Um einen Bezug zu
EDUtest herzustellen, frage ich mich also, inwieweit man es irgendwie
einbinden sollte. Auf der Hand laege ein Button, um die
privilegeareas-Benutzerfuehrung aufzurufen. Aber wie im anderen
Kommentar schon angesprochen, macht es vielleicht auch Sinn, die
Festlegung des Gebiets benutzerfreundlicher zu machen.

Originalnachricht:

**Comment by Surfer2010** [comment]
ich gebe dir recht, dass es sinnvoller ist den Mod für sich weiter bestehen zu lassen, aber wenn er "so alt" ist, wäre es dann nicht schlau ihn zu forken, evt. zu erneuern/erweitern und vor allem mit einem Namen zu belegen, der ihm diesem EDU-Project zuordnet, damit Leute die helfen wollen vllt. auch die Einzelteile finden, an denen sie weiterbasteln können?

**Comment by BiLachner** [comment]
Ja, hört sich sinnvoll an.

**Comment by zeuner** [comment]
Solange nuetzliche Updates und Erweiterungen vom Maintainer angenommen
werden, sehe ich eigentlich keinen Grund, den Code an weiterer Stelle
unabhaengig weiterzuentwickeln.

Etwas anderes vielleicht, wenn man Aenderungen hat, die spezifisch nur
fuer den EDU-Kontext nuetzlich sind. Dann macht es sicher Sinn,
separat weiterzuentwickeln. Aber ich weiss gerade nicht, wie
solche Aenderungen konkret aussehen wuerden.

Eine Zusammenstellung von Mods fuer den EDU-Einsatz als Uebersicht ist
eigentlich jenseits der Zielsetzung _dieses_ Projekts (das vor allem
technisch vereinfachte Schnittstellen fuer verschiedene Mods
implementiert). Die waere vielleicht z.B. hier ([1]) gut aufgehoben.

[1] https://github.com/minetest4edu/modpack4Edu

Originalnachricht:

**Comment by zeuner** [comment]
[English version below]

@Birgit Lachner: Man kann sich das ja noch einmal ueberlegen, wenn es
konkret Aenderungen gibt, die vom Originalprojekt nicht angenommen
werden.

-----

@Birgit Lachner: This could be reconsidered when there are any changes
which are refused by the original project.

Originalnachricht:
